[clang-tidy] Use --match-full-lines instead of --strict-whitespace in check_clang_tidy

nicovank opened the pull request:
See Discourse post here:
https://discourse.llvm.org/t/rfc-using-match-full-lines-in-clang-tidy-tests/85553

I've added `--match-partial-fixes` to all tests that were failing, unless I noticed the fix was quick and trivial. I'll do a second pass after this lands to update more tests.

nicovank commented on clang-tools-extra/test/clang-tidy/checkers/modernize/use-starts-ends-with.cpp:
```diff
@@ -172,7 +172,7 @@ void test(std::string s, std::string_view sv, sub_string ss, sub_sub_string sss,
   0 != s.compare(0, sv.length(), sv);
   // CHECK-MESSAGES: :[[@LINE-1]]:{{[0-9]+}}: warning: use starts_with
-  // CHECK-FIXES: s.starts_with(sv);
+  // CHECK-FIXES: !s.starts_with(sv);
```
This is the type of typo this can help prevent!

carlosgalvezp replied:
Nice!

carlosgalvezp commented on clang-tools-extra/test/clang-tidy/check_clang_tidy.py:
```diff
@@ -245,15 +246,15 @@ def check_no_diagnosis(self, clang_tidy_output: str) -> None:
+                self.input_file_name,
+                "--check-prefixes=" + ",".join(self.fixes.prefixes),
+            ]
+            if not self.match_partial_fixes:
```
Shouldn't it still use `-strict-whitespace` to preserve the old behavior?

nicovank replied:
Right, that makes more sense, especially since (at least one) tests work when enabled but fail when disabled. I initially thought it was strictly less aggressive to not enable it anyway but maybe not.

carlosgalvezp reviewed (approved):
LGTM!

nicovank committed:
[clang-tidy] Use --match-full-lines instead of --strict-whitespace in check_clang_tidy

See Discourse post here:
https://discourse.llvm.org/t/rfc-using-match-full-lines-in-clang-tidy-tests/85553

I've added `--match-partial-fixes` to all tests that were failing, unless I noticed the fix was quick and trivial. I'll do a second pass after this lands to update more tests.
